src/parser/meta_parser.rs

```diff
@@ -400,6 +357,18 @@ fn construct_meta_value_from_flag_array(
     data: Option<&[u8]>,
     status: Option<Status>,
 ) -> Result<MetaValue, Error> {
+    // might need something like this in the case of tombstoned data.
+    // Currently returns Some([]), but should maybe be converted to None?
+    // let data = if let Some(d) = data {
+    //     if d.is_empty() {
+    //         None
+    //     } else {
+    //         Some(d.to_vec())
+    //     }
+    // } else {
+    //     None
+    // };
+
```

Comment:
Having the MetaValue's data attribute mapped to vec![] after tombstoning (clearing the key's associated value with the x flag) doesn't seem right, we might need a small parser change to have this mapped to None instead.

Basically what's happening is that the parser currently expects a data block as part of a VA response from a meta get, but after removing the value there is simply nothing there.

Reply:
> change to have this mapped to None instead

This sounds like it makes sense